pkg/operator/operator.go

```diff
@@ -489,6 +491,20 @@ func (o *Operator) sync(key string) error {
 	return nil
 }
 
+func (o *Operator) reportError(err error, taskName string) {
```

Comment:
(nit) small suggestion to make it easier to read IMHO

```golang
func (o *Operator) reportError(err error, taskName string) {
    defer o.failedReconcileAttempts++

    klog.Infof("Operator reconciliation failed, retrying. Err: %v", err)
    if o.failedReconcileAttempts < 3 {
        // Don't update the ClusterOperator status until 3 retries have been attempted.
        return
    }

    ...
```

Reply:
Hm only that `defer` doesn't work. It expects a function call, whereas we have a statement. Not sure wrapping it in a function helps readability here.

Reply:
my bad I didn't verify that my pseudo code worked :) maybe we could still split the logs (1 for logging the error and the other telling that we update the cluster operator status)?

```golang
    klog.Infof("Reconciliation failed (%d retries). Err: %v", o.failedReconcileAttempts, err)
    if o.failedReconcileAttempts > 3 {
        // Only update the ClusterOperator status after 3 retries have been attempted to avoid flapping status.
        klog.Info("Updating ClusterOperator status to failed ")
        ...
    }
```

Reply:
Changes the logging as per your comment. I also changed the condition to 2 (we start counting at 0 so `> 3` was incorrect and to an equality check. No need to set `degraded` repeatedly I figure.